### DISCUSSION TITLE
Run rules over remote introspection schema

### DISCUSSION THREAD

**Discussion by alfa-f** [graphql-hive/graphql-eslint#1030] [Q&A] [answered]
Hi, have set up graphql-eslint locally, and can successfully run against a local .graphql file : 

`npx eslint C:\path\to\project\schema.graphql
`
yielding expected output : 

```
   5:6   error  Description is required for `type Queries`                                                                                        @graphql-eslint/require-description
   5:6   error  Queries must have exactly one non-nullable unique identifier. Accepted name: id. Accepted type: ID                                @graphql-eslint/strict-id-in-types
```

We have configured a remote schema using the parserOptions : 

```
      "parserOptions": {
        "schema": "https://my.graphql-service.com/graphql",
        "schemaOptions": {
          "headers": {
            "Authorization": "Bearer ..."
          }   
```

It appears that the remote schema is used as a 'base' that the local file is expected to extend, and may reference types from etc. However what we need is a way to run the rules against the _remote_ schema itself. Is there a way to do this without first downloading the introspection output to a local json file?

Thanks,

Alan.

**Comment by dimaMachina** [comment]
Hi Alan, good question, it's not documented but it's possible, just tested it.
To make it works just provide an array, where 1st item is your URL endpoint and 2nd item is glob pattern to your local graphql files:
```diff
      "parserOptions": {
-       "schema": "https://my.graphql-service.com/graphql",
+       "schema": ["https://my.graphql-service.com/graphql", "path-to-your-local-files/*.graphql"],
        "schemaOptions": {
          "headers": {
            "Authorization": "Bearer ..."
          }
```
Would you contribute and submit a PR with documentation change?

**Reply by alfa-f** [reply]
Hi, thanks for your reply. Tried adding the line above as per your example but still no joy. Could you possibly send me your complete eslintrc.json and the npx command you are executing to run the linter, so I can compare?

Thanks

**Comment by dimaMachina** [comment, marked as answer]
Sorry, I reread your question if you want to run `graphql-eslint` rules against your remote schema - there is no way without downloading your schema as `graphql` file. But if you want to extend your remote schema with your local schema you can do it in the way described above.
